subsys/nfc/lib/platform_internal_thread.c

```diff
@@ -251,9 +295,36 @@ void nfc_platform_cb_request(const void *ctx,
 	}
 
 end:
-#ifdef CONFIG_NFC_LOW_LATENCY_IRQ
+#ifdef CONFIG_NFC_ZERO_LATENCY_IRQ
 	NVIC_SetPendingIRQ(NFC_SWI_IRQN);
 #else
 	schedule_callback();
-#endif /* CONFIG_NFC_LOW_LATENCY_IRQ */
+#endif /* CONFIG_NFC_ZERO_LATENCY_IRQ */
+}
+
+void nfc_platform_event_handler(nrfx_nfct_evt_t const *event)
+{
+	switch (event->evt_id) {
+	case NRFX_NFCT_EVT_FIELD_DETECTED:
+#ifdef CONFIG_NFC_ZERO_LATENCY_IRQ
```

Comment:
My general suggestion is to avoid conditional compilation by using `IS_ENABLED()`

Reply:
This can't be swapped here because of clk_mode variable which is conditional.